src/context.ts

```diff
@@ -108,9 +107,9 @@ export async function runWithParentContext<R>(
   return await asyncLocalCtx.run(
     {
       ...pctx,
+      ...ctx,
       workflowTimeoutMS: undefined, // Becomes deadline
```

Comment:
@kraftp I think this should move to the callsite in runWorkflow instead of always being set here.